### PR TITLE
Class refactor, new helpers serve.add and serve.delete

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,10 +72,37 @@ const goodbye = require('graceful-goodbye')
 
 const server = http.createServer()
 const close = graceful(server)
-await serve(drive, { server })
+const serve = new ServeDrive({ server })
+// serve.add(drive)
```
Comment (HDegroote) [inline README.md:76]: Is this meant to be commented?

File: index.js [hunk 2]
```diff
@@ -1,21 +1,72 @@
...
+    } catch (err) {
+      if (!this.anyPort) throw err
+      if (err.code !== 'EADDRINUSE') throw err
+      await listen(this.server, 0, this.host)
+    }
+  }
 
-  const server = opts.server || http.createServer()
+  async _close () {
+    if (this.closed) return
```
Comment (HDegroote) [inline index.js:35]: No need for this and next line if using ready-resource

Comment (HDegroote) [reply index.js:35]: Neither for the check on this.opened I think, but not 100% sure there

Comment (LuKks) [reply index.js:35]: It's not needed, it was inherited from the previous code that was doing its own opening/closing

**Pull request merged**